File: 226.Invert-Binary-Tree/tree.py
from typing import Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root == None:
            return None

        stack = [root]

        # O(n)
        while stack:
            if node := stack.pop():
                node.left, node.right = node.right, node.left
                stack.append(node.left)
                stack.append(node.right)

        return root

File: 226.Invert-Binary-Tree/tree.c
#include <stdlib.h>

struct TreeNode {
  int val;
  struct TreeNode *left;
  struct TreeNode *right;
};

struct TreeNode *invertTree(struct TreeNode *root) {
  if (!root) {
    return NULL;
  }

  // O(n)
  struct TreeNode *tmp = root->left;
  root->left = invertTree(root->right);
  root->right = invertTree(tmp);

  return root;
}
